app/celer-sim/celer-sim.cc

```diff
@@ -131,16 +135,25 @@ void run(std::istream* is, std::shared_ptr<OutputRegistry> output)
             activate_device_local();
 
             // Run a single event on a single thread
-            CELER_TRY_HANDLE(result.events[event] = run_stream(
+            TransporterResult event_result;
+            CELER_TRY_HANDLE(event_result = run_stream(
                                  StreamId(get_openmp_thread()), EventId(event)),
                              capture_exception);
+            if (run_input->transporter_result)
+            {
+                result.events[event] = std::move(event_result);
+            }
         }
         log_and_rethrow(std::move(capture_exception));
     }
+
     result.action_times = run_stream.get_action_times();
     result.total_time = get_transport_time();
     record_mem = {};
-    output->insert(std::make_shared<RunnerOutput>(std::move(result)));
+    if (run_input->transporter_result)
+    {
+        output->insert(std::make_shared<RunnerOutput>(std::move(result)));
+    }
```

Comment:
We should still write the simulation result (timing info, number of CPU threads) either way.
```suggestion
    output->insert(std::make_shared<RunnerOutput>(std::move(result)));
```